kubernetes/controller/internal/resolution/workerpool/workerpool.go

```diff
@@ -0,0 +1,247 @@
+package workerpool
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/hashicorp/golang-lru/v2/expirable"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	descriptor "ocm.software/open-component-model/bindings/go/descriptor/runtime"
+	"ocm.software/open-component-model/bindings/go/repository"
+	"ocm.software/open-component-model/bindings/go/runtime"
+)
+
+// ResolveOptions contains all the options the resolution service requires to perform a resolve operation.
+type ResolveOptions struct {
+	RepositorySpec runtime.Typed
+	Component      string
+	Version        string
+	Key            string
+	Repository     repository.ComponentVersionRepository
+}
+
+// Result contains the result of a resolution including any errors that might have occurred.
+type Result struct {
+	Value any
```

Comment:
I think a worker pool result should be typed as well as the worker pool itself

Reply:
no you would cache the Result which contains only a certain value type right?. so [string,*Result[val]]